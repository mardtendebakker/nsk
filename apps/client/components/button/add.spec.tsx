import { fireEvent, render } from '@testing-library/react';
import Add from './add';

describe('add', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Add onClick={() => {}} title="title" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('handles click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Add onClick={onClick} />);
    fireEvent.click(getByRole('button'));

    expect(onClick).toHaveBeenCalled();
  });
  it('disables click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Add onClick={onClick} disabled />);
    fireEvent.click(getByRole('button'));

    expect(onClick).not.toHaveBeenCalled();
  });
});
