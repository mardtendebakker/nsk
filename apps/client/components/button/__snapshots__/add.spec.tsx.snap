// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add matches snapshot 1`] = `
<DocumentFragment>
  <span
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rb4sk0-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lr4af1-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;
